src/neuroglancer/datasource/graphene/backend.ts

```diff
@@ -0,0 +1,664 @@
+    }
+    else {
+      // Download shard fragments with verification (response contains size and offset)
+      fragmentDownloadPromise = getVerifiedFragmentPromise(chunk, parameters, cancellationToken);
+    }
+  } else {
+    fragmentDownloadPromise = cancellableFetchOk(
+      `${parameters.fragmentUrl}/${chunk.fragmentId}`, {}, responseArrayBuffer,
+      cancellationToken);
+  }
+  return fragmentDownloadPromise;
+}
+
+@registerSharedObject() export class GrapheneMeshSource extends
+(WithParameters(WithSharedCredentialsProviderCounterpart<SpecialProtocolCredentials>()(MeshSource), MeshSourceParameters)) {
+  protected minishardIndexSources: MinishardIndexSource[];
+
+  async download(chunk: ManifestChunk, cancellationToken: CancellationToken) {
+    const {parameters} = this;
+    let url = `${parameters.manifestUrl}/manifest`;
+    let manifestUrl = `${url}/${chunk.objectId}:${parameters.lod}?verify=1&prepend_seg_ids=1`;
+
+    // speculative manifest isn't working all the time
+    // race condition is the prime suspect so use verify=true
+    chunk.verifyFragments = true;
+
+    // parameters.sharding is a proxy for mesh format
+    // if undefined, mesh format is old else new
+    if (parameters.sharding !== undefined) {
+      chunk.manifestType = GRAPHENE_MANIFEST_SHARDED;
+      if (this.minishardIndexSources === undefined) {
+        this.minishardIndexSources = getGrapheneMinishardIndexDataSources(
+          this.chunkManager, undefined, {url: parameters.fragmentUrl, sharding: parameters.sharding})!;
+      }
+      if (!chunk.verifyFragments) {
+        manifestUrl = `${url}/${chunk.objectId}:${parameters.lod}?verify=0&prepend_seg_ids=1`;
+      }
+    }
+    await cancellableFetchSpecialOk(this.credentialsProvider, manifestUrl, {}, responseJson, cancellationToken)
+        .then(response => decodeManifestChunk(chunk, response));
+  }
+
+  async downloadFragment(chunk: FragmentChunk, cancellationToken: CancellationToken) {
+    const {parameters, minishardIndexSources} = this;
+    const fragmentDownloadPromise = getFragmentDownloadPromise(
+      chunk, parameters, minishardIndexSources, cancellationToken
+    );
+
+    const dracoModulePromise = DracoLoader.default;
+    const readyToDecode = Promise.all([fragmentDownloadPromise, dracoModulePromise]);
+    return readyToDecode.then(
+        response => {
+          try {
+            decodeDracoFragmentChunk(chunk, response[0], response[1].decoderModule);
+          } catch (err) {
+            if (err instanceof TypeError) {
+              // not a draco mesh
+              decodeFragmentChunk(chunk, response[0]);
+            }
+          }
+        },
+        error => {
+          if (error instanceof HttpError && error.status == 404) {
+            chunk.source!.removeChunk(chunk);
+          }
+          Promise.reject(error);
+        });
+  }
+}
+
+async function fetchByUint64(
```

Comment:
If possible it would be best to avoid duplication of code from precomputed/backend.ts